workitem/typegroup_repository.go

```diff
@@ -0,0 +1,165 @@
+package workitem
+
+import (
+	"context"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/spacetemplate"
+
+	"github.com/fabric8-services/fabric8-wit/application/repository"
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+// WorkItemTypeGroupRepository encapsulates storage & retrieval of work item
+// type groups
+type WorkItemTypeGroupRepository interface {
```

Comment:
type name will be used as workitem.WorkItemTypeGroupRepository by other packages, and that stutters; consider calling this TypeGroupRepository